pkg/providers/tinkerbell/reconciler/reconciler.go

```diff
@@ -239,32 +280,25 @@ func (r *Reconciler) ValidateHardware(ctx context.Context, log logr.Logger, tink
 	clusterSpec := tinkerbellScope.ClusterSpec
 	log = log.WithValues("phase", "validateHardware")
 
-	capiCluster, err := controller.GetCAPICluster(ctx, r.client, clusterSpec.Cluster)
-	if err != nil {
-		return controller.Result{}, errors.Wrap(err, "validating tinkerbell hardware")
-	}
-	if capiCluster != nil {
-		// If CAPI cluster exists, the hardware has been validated
-		// and it's possibly already in use so no need to validate it again.
-		log.V(3).Info("CAPI cluster already exists, skipping hardware validations")
-		return controller.Result{}, nil
-	}
-
 	// We need a new reader each time so that the catalogue gets recreated.
 	etcdReader := hardware.NewETCDReader(r.client)
 	if err := etcdReader.NewCatalogueFromETCD(ctx); err != nil {
 		log.Error(err, "Hardware validation failure")
 		failureMessage := err.Error()
 		clusterSpec.Cluster.Status.FailureMessage = &failureMessage
-
 		return controller.ResultWithReturn(), nil
 	}
 
 	var v tinkerbell.ClusterSpecValidator
-	v.Register(
-		tinkerbell.MinimumHardwareAvailableAssertionForCreate(etcdReader.GetCatalogue()),
-		tinkerbell.HardwareSatisfiesOnlyOneSelectorAssertion(etcdReader.GetCatalogue()),
-	)
+	v.Register(tinkerbell.HardwareSatisfiesOnlyOneSelectorAssertion(etcdReader.GetCatalogue()))
+
+	// todo: AssertionsForScaleUpDown
+	switch *tinkerbellScope.ClusterChange {
```

Comment:
Having this decision logic here makes the decision to add the validations execution context dependent (controller vs CLI).

I would think this logic would reside in some shared function, and possibly some other execution context agnostic validations like the `HardwareSatisfiesOnlyOneSelectorAssertion`.